oeps/oep-1.rst

```diff
@@ -0,0 +1,266 @@
+mailing list archive or to the appropriate section of the PR.
+
+OEPs can also be superseded by a different OEP, rendering the original
+obsolete.
+
+The possible paths of the status of OEPs are as follows:
+
+.. image:: oep-1/state-flow.png
+
+OEP Maintenance
+---------------
+
+In general, OEPs are no longer modified after they have reached the Final state.
+They can be replaced by subsequent OEPs, however.
+
+What belongs in a successful OEP?
+=================================
+Each OEP should have the following parts:
+
+*Preamble*
+  A table containing meta-data about the OEP, including the OEP number,
+  a short descriptive title, the names, and optionally the contact info for each author, etc.
+
+*Abstract*
+  a short (~200 word) description of the technical issue being addressed.
+
+*Copyright*
+  All OEPs must be shared under the `Creative Commons Attribution-ShareAlike 4.0 International License`_.
+
+.. _Creative Commons Attribution-ShareAlike 4.0 International License: https://creativecommons.org/licenses/by-sa/4.0/
+
+*Specification*
+  The technical specification should describe the syntax and
+  semantics of any new API.
+
+*Motivation*
+  The motivation is critical for OEPs that want to change Open edX. It should
+  clearly explain why the existing achictecture or process is inadequate to
+  address the problem that the OEP solves, or why Open edX would be significantly
+  improved by adopting the best practice. OEP submissions without sufficient
+  motivation may be rejected outright.
+
+*Rationale*
+  The rationale fleshes out the specification by describing what motivated the
+  design and why particular design decisions were made. It should describe
+  alternate designs that were considered and related work, e.g. how the
+  feature is supported in other systems.
+
+  The rationale should provide evidence of consensus within the community
+  and discuss important objections or concerns raised during discussion.
+
+*Backwards Compatibility*
+  All OEPs that introduce backwards incompatibilities must include a section
+  describing these incompatibilities and their severity. The OEP must explain
+  how the author proposes to deal with these incompatibilities. OEP submissions
+  without a sufficient backwards compatibility treatise may be rejected outright.
+
+*Reference Implementation*
+  The reference implementation must be completed before any OEP is given status
+  "Final", but it need not be completed before the OEP is accepted. While there
+  is merit to the approach of reaching consensus on the specification and rationale
+  before writing code, the principle of "rough consensus and running code" is still
+  useful when it comes to resolving many discussions of API details.
+
+  The final implementation must include test code and documentation, following the
+  `Open edX Contribution Guidelines`_.
+
+.. _Open edX Contribution Guidelines: http://edx.readthedocs.org/projects/edx-developer-guide/en/latest/process/index.html
+
```

Comment:
A useful thing for keeping architecture on track is "Rejected Alternatives".  It can prevent revisiting decisions.